### PR TITLE
fix map erase

### PR DESCRIPTION
**Pull request by panda-sheep**
As title.
Thanks chaos for checking out this problem.

### CODE REVIEW

**Review by critical27 — COMMENTED** [review]
image

Good job, I found this as well, we don't need `isErase` actually. Close #270

**Comment by panda-sheep** [comment]
> image
> 
> Good job, I found this as well, we don't need `isErase` actually. Close #270

Yes, this kind of reform is also ok，  see https://github.com/vesoft-inc/nebula-cpp/pull/8

**Review by critical27 — APPROVED** [review]

File: src/meta/processors/admin/Balancer.cpp [hunk 1]
```diff
@@ -752,25 +754,33 @@ int32_t Balancer::giveupLeaders(HostParts& leaderParts,
...
-            it = sourceLeaders.erase(it);
-            targetLeaders.emplace_back(partId);
-            plan.emplace_back(spaceId, partId, source, *target);
-            LOG(INFO) << "giveup plan trans leader space: " << spaceId
-                      << " part: " << partId
-                      << " from " << source.host << ":" << source.port
-                      << " to " << target->host << ":" << target->port;
-            ++taskCount;
+        if (target != targets.end()) {
+            auto& targetLeaders = leaderParts[*target];
```
Comment (liuyu85cn) [inline src/meta/processors/admin/Balancer.cpp:760]: I'm not sure if leaderParts[*target] may trigger re-hash, and invalid everything

**Review by liuyu85cn — APPROVED** [review]
good job.

**Pull request merged**